arrow-core-data/src/main/kotlin/arrow/core/Eval.kt

```diff
@@ -242,6 +267,125 @@ sealed class Eval<out A> : EvalOf<A> {
 
       return curr.value() as A
     }
+
+    fun <A, B, C> mapN(
+      a: Eval<A>,
+      b: Eval<B>,
+      map: (A, B) -> C
+    ): Eval<C> =
+      mapN(a, b, Unit, Unit, Unit, Unit, Unit, Unit, Unit) { aa, bb, _, _, _, _, _, _, _ -> map(aa, bb) }
+
+    fun <A, B, C, D> mapN(
+      a: Eval<A>,
+      b: Eval<B>,
+      c: Eval<C>,
+      map: (A, B, C) -> D
+    ): Eval<D> =
+      mapN(a, b, c, Unit, Unit, Unit, Unit, Unit, Unit, Unit) { aa, bb, cc, _, _, _, _, _, _, _ -> map(aa, bb, cc) }
```

Comment:
nit:
with the kotlin naming conventions wouldn't we define `Unit` starting with a lowercase letter same there too
https://github.com/arrow-kt/arrow-core/blob/01781b8380401f0f1d2bc60edb234bcb23aec41b/arrow-core-data/src/main/kotlin/arrow/core/Eval.kt#L175-L183